roc/context_test.go

```diff
@@ -1,36 +1,38 @@
 package roc
 
-import "testing"
+import (
+	"testing"
 
-func Test_roc_context_open(t *testing.T) {
+	"github.com/stretchr/testify/require"
+)
+
+func TestContext_Open(t *testing.T) {
 	tests := []struct {
-		config  ContextConfig
-		wantErr error
+		name   string
+		ok     bool
+		config ContextConfig
 	}{
 		{
-			config:  ContextConfig{MaxPacketSize: 50, MaxFrameSize: 70},
-			wantErr: nil,
+			name:   "ok",
+			ok:     true,
+			config: ContextConfig{MaxPacketSize: 50, MaxFrameSize: 70},
 		},
 	}
 
 	for _, tt := range tests {
-		ctx, err := OpenContext(&tt.config)
-
-		if err != tt.wantErr {
-			fail(tt.wantErr, err, t)
-		}
-
-		if err != nil {
-			continue
-		}
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, err := OpenContext(tt.config)
 
-		if ctx == nil {
-			fail("Context initialized", "Contenxt is nil", t)
-		}
+			if tt.ok {
+				require.NoError(t, err)
+				require.NotNil(t, ctx)
 
-		err = ctx.Close()
-		if err != nil {
-			fail(nil, err, t)
-		}
+				err = ctx.Close()
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				require.Nil(t, ctx)
```

Comment:
I think checking for a specific error provides for more clarity. Resorting to general errors with a string inside should be the case only if the error is unexpected. This way we can handle different types of errors and completely panic only in this case of unexpected unidentified error.

Reply:
Good point! Reworked all tests to check for specific errors.